packages/roosterjs-editor-core/lib/coreApi/createPasteFragment.ts

```diff
@@ -52,7 +52,7 @@ export const createPasteFragment: CreatePasteFragment = (
     let doc: HTMLDocument;
 
     // Step 2: Fill the BeforePasteEvent object, especially the fragment for paste
-    if (!pasteAsText && !text && imageDataUri) {
+    if (!pasteAsText && !text && imageDataUri && !rawHtml) {
```

Comment:
I'm concerned that this may bring some other issue since it is possible that we still want to paste image when there is HTML which just has the IMG tag to the image file.

Actually we have code to handle pasting from PowerPoint here: https://github.com/microsoft/roosterjs/blob/master/packages/roosterjs-editor-plugins/lib/plugins/Paste/pptConverter/convertPastedContentFromPowerPoint.ts

However, in this case it can't work because we have a check to only do this for PowerPoint: https://github.com/microsoft/roosterjs/blob/fc2dcd6f900a1ca93915da73bf189771920b1d58/packages/roosterjs-editor-plugins/lib/plugins/Paste/Paste.ts#L78

And because some code bug in createPasteFragment, we didn't retrieve the correct attribute for PPT, so that the function to handle PPT paste is not executed. You need to make the code https://github.com/microsoft/roosterjs/blob/master/packages/roosterjs-editor-core/lib/coreApi/createPasteFragment.ts#L67-L75  always executed when possible, so that the related attribute (as well as meta tag) can be parsed correctly.

Reply:
I was debugging and it seems like the rawHtml when performing the paste, do have meta Tags but are not equal to "POWERPOINT_ATTRIBUTE_VALUE", this is the rawHtml from the clipboardData,:

`
<html xmlns:m=\"http://schemas.microsoft.com/office/2004/12/omml\"\r\nxmlns=\"http://www.w3.org/TR/REC-html40\">\r\n\r\n
  <head>\r\n
    <meta http-equiv=Content-Type content=\"text/html; charset=utf-8\">\r\n
    <meta name=ProgId content=PowerPoint.Slide>\r\n
    <meta name=Generator content=\"Microsoft PowerPoint 15\">\r\n`

after parsing the Html from the string the meta tags are lost:

![image](https://user-images.githubusercontent.com/8291124/130247679-a1a4337e-9ce6-481c-a698-a65b780d9af2.png)

I tried this also copying and pasting content from word and I got the same error, the meta tags of the rawHtml are not included in the parsed Html.

I think we could do some manipulation on the rawHtml to retrieve the tags from when since the DomParser().parseFromString is not returning the meta tags.

Any feedback?

Reply:
POWERPOINT_ATTRIBUTE_VALUE  is the const name of PowerPoint.Slide:
const POWERPOINT_ATTRIBUTE_VALUE = 'PowerPoint.Slide';

See https://github.com/microsoft/roosterjs/blob/1a1b87916f2e5253da363c1e51f015626aadbe7d/packages/roosterjs-editor-plugins/lib/plugins/Paste/Paste.ts#L19

The content is right, and the meta value is right, too. The problem is the code to retrieve the meta value is not run.

Reply:
After some debugging, I noticed that the HEAD Tag is being removed from the Html when sanitizing the rawHtml (line) for instance, this is the rawHtml that is provided in the ClipboardData...
![image](https://user-images.githubusercontent.com/8291124/130273204-8a20639d-2141-4ac1-8910-a69a1dd5fd55.png)


but after Sanitizing this string in core.trustedHTMLHandler(rawHtml), only the Table tag is returned and the Head is removed.... this is the result after running trustedHTMLHandler:

![image](https://user-images.githubusercontent.com/8291124/130273043-17788f9f-94b6-45a2-9c45-e8d35f2e9d4e.png)

Reply:
Please try let DOMPurify allow Head tag similar with: https://stackoverflow.com/questions/60299226/how-to-allow-an-iframe-tag-in-dompurify-including-all-of-its-attributes 
by changing https://github.com/microsoft/roosterjs/blob/master/demo/scripts/utils/trustedHTMLHandler.ts

If that does not work, try remove DOMPurify and directly return the source HTML in turstedHTMLHandler.

Reply:
With this DOMPurity Configuration the head and the meta tags are now included to the sanitized html

![image](https://user-images.githubusercontent.com/8291124/130289235-1e32dd90-71e8-4c7a-9f38-4b5803e4bdaa.png)

Thanks